statistics/selectivity.go

```diff
@@ -283,6 +285,52 @@ func (coll *HistColl) Selectivity(ctx sessionctx.Context, exprs []expression.Exp
 			ret *= selectionFactor
 		}
 	}
+
+	// Now we try to cover those still not covered DNF conditions using independence assumption,
+	// i.e., sel(condA or condB) = sel(condA) + sel(condB) - sel(condA) * sel(condB)
+	if mask > 0 {
+		for i, expr := range remainedExprs {
+			if mask&(1<<uint64(i)) == 0 {
+				continue
+			}
+			scalarCond, ok := expr.(*expression.ScalarFunction)
+			// Make sure we only handle DNF condition.
+			if !ok || scalarCond.FuncName.L != ast.LogicOr {
+				continue
+			}
+			dnfItems := expression.FlattenDNFConditions(scalarCond)
+			dnfItems = ranger.MergeDNFItems4Col(ctx, dnfItems)
+
+			selectivity := 0.0
+			for _, cond := range dnfItems {
+				_, ok := cond.(*expression.CorrelatedColumn)
```

Comment:
Better to add a comment here to explain the the logic behind, i.e, why we do this for column, constant, and correlated column.

Reply:
Added some comments to try to explain. Thanks.